src/main/java/io/nats/client/support/Encoding.java

```diff
@@ -200,4 +200,32 @@ public static void jsonEncode(StringBuilder sb, String s) {
             }
         }
     }
+
+    public static String uriDecode(String source) {
```

Comment:
This method doesn't handle UTF-8 properly, could it be simplified to just call `URLDecoder.decode(source.replace("+", "%2B"))` ?

```
String test = "+" + URLEncoder.encode("😀") + "+";
System.out.println(test);
// +%F0%9F%98%80+
System.out.println(uriDecode("+%F0%9F%98%80+"));
// +ð+
System.out.println(URLDecoder.decode("+%F0%9F%98%80+".replace("+", "%2B")));
// +😀+
```

Reply:
changed to support unicode